test/server-unit/db.spec.js

```diff
@@ -0,0 +1,43 @@
+const db = require('../../server/lib/db');
+const { expect } = require('chai');
+var _ = require('lodash');
+
+var dbTest = function () {
+
+  before(function () {
+    var params = {
+      host     : 'localhost',
+      user     : 'bhima',
+      password : 'HISCongo2013',
+      database : 'bhima_test',
+    };
+    db.init(params);
+
+  });
+
+  it('Should check the connection to mysql', function (done) {
+    var fx = db.pool.getConnection(function (err, result) {
```

Comment:
For callbacks, the best way to test them is with "spies".  There's a chai library for it called [chai-spies](https://github.com/chaijs/chai-spies).

I'll work on this once the branch gets merged as an example.